bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,39 @@
 # frozen_string_literal: true
 
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        SharedHelpers.run_helper_subprocess(
+          command: helper_path(bundler_version: bundler_version),
+          function: function,
+          args: args,
+          env: {
+            # Bundler will pick the matching installed major version
+            "BUNDLER_VERSION" => bundler_version,
+            # Force bundler to use the helper Gemfile that has been bundled with
+            # v1, otherwise it will point to core's bundler/Gemfile which will
+            # be bundled with v2 once it's installed
+            "BUNDLE_GEMFILE" => File.join(versioned_helper_path(bundler_version: bundler_version), "Gemfile"),
+            # Unset ruby env set by running dependabot-core with bundle exec,
+            # forcing bundler to reset them from helpers/v1
+            "RUBYLIB" => nil,
+            "RUBYOPT" => nil,
+            "GEM_PATH" => nil,
+            "GEM_HOME" => nil
```

Comment:
This seems like a fair workaround, it feels like the cost of doing business since we're running bundler to run core to run bundler ♻️  😅